test/ppx_expect/main.ml

```diff
@@ -63,9 +63,18 @@ let () =
     |> List.map Filename.chop_extension
   in
   Sys.chdir test_directory;
-  try
-    List.iter run_test test_cases
-  with
-  | Failure f ->
-    prerr_endline f;
-    exit 1
+  let only_if () =
+    Sys.cygwin = false && Sys.win32 = false &&
+    (* 4.02.3 prints file paths differently *)
+    Scanf.sscanf Sys.ocaml_version "%u.%u"
+      (fun major minor -> major >= 4 && minor >= 3)
```

Comment:
I think 5.00 will wrongly fail this test.

Reply:
Good point. Note that we'll have a similiar problem in https://github.com/ocsigen/lwt/blob/b52e2115c4b42136fa9480939135ababa219631b/src/util/configure.ml#L100

Reply:
Good catch, and thanks for noting. I'll push to `master` to fix that.